### ISSUE TITLE
Cannot connect to localhost with DynamoDBClient "EndpointError: Endpoint URL must be a string, got object"

### ISSUE THREAD

**Issue by allanvobraun** [aws/aws-sdk-js-v3#4095]
### Checkboxes for prior research

- [X] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest)
- [X] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js).
- [X] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js-v3/issues) and didn't find any solution.

### Describe the bug

With this configuration:
```
this.dynamoDB = DynamoDBClient({
      region: 'us-east-1',
      endpoint: 'http://localhost:5000/',
      credentials: {
        accessKeyId: 'xxx',
        secretAccessKey: 'yyy',
      },
    });
```
When executing a dynamodb command like:
```
this.dynamoDB.send(
      new ScanCommand({
        TableName: this.constraintTableName,
      }),
    );
```
I got this error:
```
EndpointError: Endpoint URL must be a string, got object
    at getEndpointUrl (/home/allanbraun/fontes/cnm-sphinx/.esbuild/.build/node_modules/@aws-sdk/util-endpoints/dist-cjs/utils/getEndpointUrl.js:19:11)
    at evaluateEndpointRule (/home/allanbraun/fontes/cnm-sphinx/.esbuild/.build/node_modules/@aws-sdk/util-endpoints/dist-cjs/utils/evaluateEndpointRule.js:26:50)
    at evaluateRules (/home/allanbraun/fontes/cnm-sphinx/.esbuild/.build/node_modules/@aws-sdk/util-endpoints/dist-cjs/utils/evaluateRules.js:11:89)
    at evaluateTreeRule (/home/allanbraun/fontes/cnm-sphinx/.esbuild/.build/node_modules/@aws-sdk/util-endpoints/dist-cjs/utils/evaluateTreeRule.js:12:46)
    at evaluateRules (/home/allanbraun/fontes/cnm-sphinx/.esbuild/.build/node_modules/@aws-sdk/util-endpoints/dist-cjs/utils/evaluateRules.js:20:81)
    at evaluateTreeRule (/home/allanbraun/fontes/cnm-sphinx/.esbuild/.build/node_modules/@aws-sdk/util-endpoints/dist-cjs/utils/evaluateTreeRule.js:12:46)
    at evaluateRules (/home/allanbraun/fontes/cnm-sphinx/.esbuild/.build/node_modules/@aws-sdk/util-endpoints/dist-cjs/utils/evaluateRules.js:20:81)
    at evaluateTreeRule (/home/allanbraun/fontes/cnm-sphinx/.esbuild/.build/node_modules/@aws-sdk/util-endpoints/dist-cjs/utils/evaluateTreeRule.js:12:46)
    at evaluateRules (/home/allanbraun/fontes/cnm-sphinx/.esbuild/.build/node_modules/@aws-sdk/util-endpoints/dist-cjs/utils/evaluateRules.js:20:81)
    at resolveEndpoint (/home/allanbraun/fontes/cnm-sphinx/.esbuild/.build/node_modules/@aws-sdk/util-endpoints/dist-cjs/resolveEndpoint.js:26:48)
```

### SDK version number

@aws-sdk/client-dynamodb: "^3.196.0",

### Which JavaScript Runtime is this issue in?

Node.js

### Details of the browser/Node.js/ReactNative version

v16.17.1

### Reproduction Steps

Setup a local dynamodb instance and config a client like this:
```
 DynamoDBClient({
      region: 'us-east-1',
      endpoint: 'http://localhost:5000/',
      credentials: {
        accessKeyId: 'xxx',
        secretAccessKey: 'yyy',
      },
    });
```

### Observed Behavior

EndpointError: Endpoint URL must be a string, got object
    at getEndpointUrl (/home/allanbraun/fontes/cnm-sphinx/.esbuild/.build/node_modules/@aws-sdk/util-endpoints/dist-cjs/utils/getEndpointUrl.js:19:11)
    at evaluateEndpointRule (/home/allanbraun/fontes/cnm-sphinx/.esbuild/.build/node_modules/@aws-sdk/util-endpoints/dist-cjs/utils/evaluateEndpointRule.js:26:50)
    at evaluateRules (/home/allanbraun/fontes/cnm-sphinx/.esbuild/.build/node_modules/@aws-sdk/util-endpoints/dist-cjs/utils/evaluateRules.js:11:89)
    at evaluateTreeRule (/home/allanbraun/fontes/cnm-sphinx/.esbuild/.build/node_modules/@aws-sdk/util-endpoints/dist-cjs/utils/evaluateTreeRule.js:12:46)
    at evaluateRules (/home/allanbraun/fontes/cnm-sphinx/.esbuild/.build/node_modules/@aws-sdk/util-endpoints/dist-cjs/utils/evaluateRules.js:20:81)
    at evaluateTreeRule (/home/allanbraun/fontes/cnm-sphinx/.esbuild/.build/node_modules/@aws-sdk/util-endpoints/dist-cjs/utils/evaluateTreeRule.js:12:46)
    at evaluateRules (/home/allanbraun/fontes/cnm-sphinx/.esbuild/.build/node_modules/@aws-sdk/util-endpoints/dist-cjs/utils/evaluateRules.js:20:81)
    at evaluateTreeRule (/home/allanbraun/fontes/cnm-sphinx/.esbuild/.build/node_modules/@aws-sdk/util-endpoints/dist-cjs/utils/evaluateTreeRule.js:12:46)
    at evaluateRules (/home/allanbraun/fontes/cnm-sphinx/.esbuild/.build/node_modules/@aws-sdk/util-endpoints/dist-cjs/utils/evaluateRules.js:20:81)
    at resolveEndpoint (/home/allanbraun/fontes/cnm-sphinx/.esbuild/.build/node_modules/@aws-sdk/util-endpoints/dist-cjs/resolveEndpoint.js:26:48)

### Expected Behavior

Expect to dynamodb connect to the local instance with the localhost url.
(works on v2)

### Possible Solution

util-endpoints/dist-cjs/utils/getEndpointUrl.ts
getEndpointUrl only accept strings, but for some reason a object is passed to the function as config options.

### Additional Information/Context

using esbuild, for a localsetup with serverless framework and serverless offline

**Comment by magic-m-johnson** [comment]
same issue, this is a major bug. please fix asap.

**Comment by ChristopheCubat** [comment]
I confim the issue, it was working until version `3.192.0` at least

**Comment by kuhe** [comment]
Thanks for reporting this issue, a fix was included in today's release of version [3.197.0](https://github.com/aws/aws-sdk-js-v3/releases/tag/v3.197.0).

**Comment by magic-m-johnson** [comment]
> Thanks for reporting this issue, a fix was included in today's release of version [3.197.0](https://github.com/aws/aws-sdk-js-v3/releases/tag/v3.197.0).

Thanks, updated and confirmed working.
